mm2src/mm2_metrics/src/native.rs

```diff
@@ -0,0 +1,509 @@
+use common::log::error;
+use common::{executor::{spawn, Timer},
+             log::{LogArc, LogWeak}};
+use metrics::{Key, Label};
+use metrics_exporter_prometheus::PrometheusBuilder;
+use mm2_err_handle::prelude::MmError;
+use serde_json::Value;
+use std::{collections::HashMap,
+          sync::{atomic::Ordering, Arc}};
+
+use crate::{common::log::Tag,
+            recorder::{labels_to_tags, name_value_map_to_message},
+            MetricsOps, MmRecorder};
+use crate::{MmMetricsError, MmMetricsResult};
+
+const QUANTILES: &[f64] = &[0.0, 1.0];
+
+type MetricLabels = Vec<Label>;
+
+pub(crate) type MetricNameValueMap = HashMap<String, PreparedMetric>;
+
+/// Construct Vec<Label> from a slice of strings.
+#[macro_export]
+macro_rules! mm_label {
+    ($($label_key:expr => $label_val:expr),+) => {{
+         let labels = vec![$(($label_key.to_owned(), $label_val.to_owned())),+];
+         labels
```

Comment:
According to this conversation https://github.com/KomodoPlatform/atomicDEX-API/pull/1255#discussion_r864281984, please try to avoid adding extra variables unless it makes sense from a readability point of view